keras/utils/test_utils.py

```diff
@@ -92,6 +92,7 @@ def layer_test(layer_cls, kwargs={}, input_shape=None, input_dtype=None,
     # check shape inference
     model = Model(x, y)
     expected_output_shape = layer.compute_output_shape(input_shape)
+    model.compile('rmsprop', 'mse')
```

Comment:
The Keras API does not require `compile` before calling `predict`, because compile merely configures training and is not related to inference. If MXNet requires it, that's a bug and it should be fixed.

Reply:
This is fixed.